spec.html

```diff
@@ -31538,12 +31398,14 @@
             1. Let _srcBuffer_ be the value of _O_'s [[ViewedArrayBuffer]] internal slot.
             1. If IsDetachedBuffer(_srcBuffer_) is *true*, throw a *TypeError* exception.
             1. Let _targetBuffer_ be the value of _A_'s [[ViewedArrayBuffer]] internal slot.
+            1. If IsDetachedBuffer(_targetBuffer_) is *true*, throw a *TypeError* exception."
+            1. If the value of _A_'s [[ArrayLength]] internal slot < _count_, throw a *TypeError* exception.
             1. Let _elementSize_ be the Number value of the Element Size value specified in <emu-xref href="#table-49"></emu-xref> for _srcType_.
             1. NOTE: If _srcType_ and _targetType_ are the same the transfer must be performed in a manner that preserves the bit-level encoding of the source data.
             1. Let _srcByteOffet_ be the value of _O_'s [[ByteOffset]] internal slot.
-            1. Let _targetByteIndex_ be 0.
+            1. Let _targetByteIndex_ be _A_'s [[ByteOffset]] internal slot.
             1. Let _srcByteIndex_ be (_k_ &times; _elementSize_) + _srcByteOffet_.
-            1. Repeat, while _targetByteIndex_ &lt; _count_ &times; _elementSize_
+            1. Repeat, while _targetByteIndex_ &minus; _A_'s [[ByteOffset]] internal slot &lt; _count_ &times; _elementSize_
```

Comment:
For consistency with other loops (e.g. %TypedArray%.prototype.set), I'd prefer:

1. Let limit be targetByteIndex + count &times; elementSize.
1. Repeat, while targetByteIndex &lt; limit